cEP-0017.md

```diff
@@ -0,0 +1,356 @@
+        for bear in bears:
+            for requirement in bear.REQUIREMENTS:
+                print(requirement.package, requirement.version)
+
+        choice = raw_input().lower()
+        if  choice  in  consts.TRUE_STRINGS:
+            return   True
+        elif  choice  in  consts.FALSE_STRINGS:
+            return   False
+        else :
+            sys.stdout.write( "Please respond with 'yes' or 'no'" )
+```
+
+### Splitting `DistributionRequirement`
+
+```
+from coala_utils.decorators import generate_eq, generate_repr
+from sarge import run, Capture
+
+
+@generate_eq("platform", "package", "version", "repo")
+@generate_repr()
+class PlatformRequirement:
+    """
+    This class helps keeping track of bear distribution requirements.
+    """
+
+    REQUIREMENTS = {}
+
+    def __init__(self, platform: str, package: str, version="", repo=""):
+        """
+        :param platform:  A string with the name of the distribution
+        :param package: A string with the name of the package to be installed.
+        :param version: A version string. Leave empty to specify latest version.
+        :param repo:    The repository from which the package is to be
+                        installed.
+        """
+        self.platform = platform
+        self.package = package
+        self.version = version
+        self.repo = repo
+
+    def __str__(self):
+        """
+        Just return package name, followed by version if given.
+        """
+        if self.version:
+            return "{}{} on {}".format(self.package,
+                                       self.version,
+                                       self.platform)
+        else:
+            return "{} on {}".format(self.package,
+                                     self.platform)
+
+    def install_package(self):
+        """
+        Runs the install command for the package given in a sub-process.
+        """
+        run(" ".join(self.install_command()), stdout=Capture(),
+            stderr=Capture())
+
+    def install_command(self):
+        """
+        Returns a string with the installation command, to be used by
+        "install_package()".
+
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    def is_installed(self):
+        """
+        Check if the requirement is satisfied.
+
+        :return: Returns True if satisfied, False if not.
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    def check_platform():
+        """
+        Check if the user is on the target platform
+
+        :return: Returns True if user is on correct platform for this Requirement,
+        otherwise false.
+        """
+        if platform.system() is 'Darwin':
+            return self.platform == 'Darwin'
+        elif platform.system() is 'Linux':
+            distro = platform.linux_distribution()[0]
```

Comment:
eww ... this will fail badly ... use https://pypi.org/project/distro

that will eventually also replace the 'Darwin' case as well

and this should be cached so that multiple subclasses does not involve detecting/interrogating platform info multiple times.